File: gntl_app_backend/src/main/java/com/gentlemanstore/user/dto/UserDTO.java
package com.gentlemanstore.user.dto;

import com.gentlemanstore.user.model.Address;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDTO {
    private Long id;
    private String email;
    private String firstName;
    private String lastName;
    private String phoneNumber;
    private LocalDateTime createdAt;
    private List<AddressDTO> addresses;
    private String role;
    private Boolean deleted;
}

File: gntl_app_frontend/app/src/main/java/com/gentlemanstore/feature/profile/data/dto/UserDto.kt
package com.gentlemanstore.feature.profile.data.dto

data class AddressInUser(
    val id: Long,
    val street: String,
    val apartment: String?,
    val city: String,
    val postalCode: String,
    val country: String,
    val isDefault: Boolean
)

data class UserResponse(
    val id: Long,
    val email: String,
    val firstName: String,
    val lastName: String,
    val phoneNumber: String?,
    val createdAt: String,
    val addresses: List<AddressInUser>?
)

data class UpdateUserRequest(
    val firstName: String,
    val lastName: String,
    val phoneNumber: String?
)